middleware.js

```
@@ -26,23 +27,39 @@ module.exports.isAdminAuthenticated = (req, res, next) => {
   }
 }
 
-module.exports.upload = function (id, required = true) {
+module.exports.upload = (id) => {
   const impl = upload.single(id);
   return async (req, res, next) => {
-    if (required || req.body[id]) {
-      impl(req, res, async (err) => {
-        if (err instanceof multer.MulterError) {
-          req.flash('error', 'Please upload a valid image. Only JPEG, JPG, and PNG files are allowed, and they must be under 5MB.');
-          res.redirect(req.url);
-        } else if (err) {
-          req.flash('error', 'An error occurred while trying to upload your image! Please try again. If the issue persists, contact us.');
-          res.redirect(req.url);
-        } else {
-          next();
+    impl(req, res, async (err) => {
+      if (err instanceof multer.MulterError) {
+        req.flash('error', 'Please upload a valid image. Only JPEG, JPG, and PNG files are allowed, and they must be under 5MB.');
+        res.redirect(req.url);
+      } else if (err) {
+        req.flash('error', 'An error occurred while trying to upload your image! Please try again. If the issue persists, contact us.');
+        res.redirect(req.url);
+      } else {
+        next();
+      }
+    });
+  }
+}
+
+module.exports.fallible = (block) => {
+  return async (req, res, next) => {
+    try {
+      await block(req, res);
+    } catch (e) {
+      // If any upload middleware was used, we will need to remove the
+      // newly-uploaded image as it's unlikely to be referenced anywhere
+      // in the database.
+      if (req.file) {
+        try {
+          fs.unlinkSync("uploads/" + req.file.filename)
+        } catch (e) {
+          // Just ignore this.
```

Why is this catch ignored? In the rare case that you have an empty catch, you should at least explain it

---

If we let this error be thrown, the server will crash. If we route it to `next`, we would override the error we wanted to catch in the first place and make debugging harder. I'll clarify this.